commit 156bea38597ced5de7521ada2e85fb67aead21db
Author: Glenn Strauss <gstrauss@gluelogic.com>
Date:   Tue Aug 2 22:32:28 2016 -0400

    [TLS] SSL_shutdown() only if handshake finished
    
    avoid noise in logs due to calling SSL_shutdown() on a connection
    that has not yet completed TLS handshake

diff --git a/src/connections.c b/src/connections.c
index ccdf360..4cd2139 100644
--- a/src/connections.c
+++ b/src/connections.c
@@ -183,7 +183,7 @@ static void connection_handle_shutdown(server *srv, connection *con) {
 
 #ifdef USE_OPENSSL
 	server_socket *srv_sock = con->srv_socket;
-	if (srv_sock->is_ssl) {
+	if (srv_sock->is_ssl && SSL_is_init_finished(con->ssl)) {
 		int ret, ssl_r;
 		unsigned long err;
 		ERR_clear_error();
